hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/security/OzoneTokenIdentifier.java

```diff
@@ -77,6 +80,122 @@ public Text getKind() {
     return KIND_NAME;
   }
 
+  /** Instead of relying on proto serialization, this
+   *  provides  explicit serialization for OzoneTokenIdentifier.
+   * @return byte[]
+   */
+  public byte[] toUniqueSerializedKey() {
+    ByteBuffer result =
+        ByteBuffer.allocate(4096);
+    result.order(ByteOrder.BIG_ENDIAN);
+    try {
+      result.putLong(getIssueDate());
+      result.putInt(getMasterKeyId());
+      result.putInt(getSequenceNumber());
+
+      result.putLong(getMaxDate());
+
+      result.putInt(getOwner().toString().length());
+      result.put(getOwner().toString().getBytes(StandardCharsets.UTF_8));
+
+      result.putInt(getRealUser().toString().length());
+      result.put(getRealUser().toString().getBytes(StandardCharsets.UTF_8));
+
+      result.putInt(getRenewer().toString().length());
+      result.put(getRenewer().toString().getBytes(StandardCharsets.UTF_8));
+
+      result.putInt(getTokenType().getNumber());
+      // Set s3 specific fields.
+      if (getTokenType().equals(S3AUTHINFO)) {
+        result.putInt(getAwsAccessId().length());
+        result.put(getAwsAccessId().getBytes(StandardCharsets.UTF_8));
+
+        result.putInt(getSignature().length());
+        result.put(getSignature().getBytes(StandardCharsets.UTF_8));
+
+        result.putInt(getStrToSign().length());
+        result.put(getStrToSign().getBytes(StandardCharsets.UTF_8));
+      } else {
+        result.putInt(getOmCertSerialId().length());
+        result.put(getOmCertSerialId().getBytes(StandardCharsets.UTF_8));
+        if (getOmServiceId() != null) {
+          result.putInt(getOmServiceId().length());
+          result.put(getOmServiceId().getBytes(StandardCharsets.UTF_8));
+        } else {
+          result.putInt(0);
+        }
+      }
+    } catch (IndexOutOfBoundsException e) {
+      throw new IllegalArgumentException(
+          "Can't encode the the raw data ", e);
+    }
+    return result.array();
+  }
+
+  /** Instead of relying on proto deserialization, this
+   *  provides  explicit deserialization for OzoneTokenIdentifier.
+   * @return byte[]
+   */
+  public static OzoneTokenIdentifier fromUniqueSerializedKey(byte[] rawData) {
+    OzoneTokenIdentifier result = newInstance();
```

Comment:
We could simplify the code by leveraging the readFields from parent class to deserialize non-protoc token id.

Reply:
Earlier I wanted to do explicit serialization, but your suggestion looks simpler. I will try this.

Reply:
I have uploaded the changes as suggested by @xiaoyuyao Please take a look..

Reply:
One failure in coverage check looks unreleted.